__tests__/models/queue.test.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import Queue from '../../src/models/queue';
+
+test('Queue operations', () => {
+    const q = new Queue();
+    expect(q.size()).toEqual(0);
+    const elements = [1, 2, 3];
+    // Enqueue elements
+    elements.forEach(el => q.enqueue(el));
+    expect(q.size()).toEqual(3);
+    // dequeued elements should in the same order as they were enqueued
+    elements.forEach(expected => {
+        // Peek should return the next element
+        const peekedEl = q.peek();
+        expect(peekedEl).toEqual(expected);
```

Comment:
Check the size after peeking and after dequeuing.

Reply:
Done

Reply:
Thanks!